### ISSUE TITLE
Empty frame list for nonexistent ID3v2 frame

### ISSUE THREAD

**Issue by silverhammermba** [taglib/taglib#719]
Taglib says that this [mp3 file](https://www.dropbox.com/s/qi9osae9pwxtmq9/crash.mp3?dl=0) has a ID3v2 tag containing a `TDRC` frame. *But* the frame list it returns is empty, and looking at the file in a hexeditor, I don't see a `TDRC` in the tag.

This is what was causing #707 ([Kodi's bug report](http://trac.kodi.tv/ticket/16463)). Now maybe it's Kodi's fault for assuming that a frame list will not be empty (if so, let me know so I can file a bug there), but why is taglib returning an empty list in the first place?

**Comment by TsudaKageyu** [comment]
Because some functions like `ID3v2::Tag::frameList()` returns the **reference**  to a list/map.

For example, if you do like this:
```
MPEG::File f("xxx.mp3");
auto list = f.ID3v2Tag()->frameList("ABCD"); // "ABCD" frame doesn't exist in the tag.
```
`frameList()` has to return a reference even if the list is empty, and an empty list has to be held in the tag object.

**Comment by silverhammermba** [comment]
But my code looks like this:

```c++
for (const auto& pair : id3v2->frameListMap())
{
	std::cout << pair.first << ": " << pair.second.size() << std::endl;
}
```

and my output is

```
COMM: 1
TALB: 1
TBPM: 1
TCON: 1
TCOP: 1
TDRC: 0
TENC: 1
TIT2: 1
TOPE: 1
TPE1: 1
TPE2: 1
TRCK: 1
UFID: 1
WXXX: 1
```

I get that if I explicitly ask for a missing frame it would return an empty list, but how could there be an empty frame when I'm just iterating through the tag?

**Comment by lalinsky** [comment]
I guess the empty frame is there because of this code:

https://github.com/taglib/taglib/blob/a0b86836568e294a6f6e83697ee6f9b1b8efb5cf/taglib/mpeg/id3v2/id3v2framefactory.cpp#L336

It wants to check if the frame is there, but as a side-effect it creates the empty entry.

**Comment by silverhammermba** [comment]
Yeah, I think that's it. I tried adding `id3v2->frameList("TDAT")` to my code before the loop and then it also showed up when I iterated (with size 0).

So maybe taglib needs something like `TagLib::ID3v2::Tag::hasFrame` that can search for frames without modifying the frame list. As-is, it seems like the `TDRC` is being accidentally created as a side-effect.

**Comment by ufleisch** [comment]
You can use `frameListMap().value("TDRC").size()` instead.
